src/main/java/com/plugin/HttpOutput.java

```diff
@@ -20,37 +28,85 @@
 import com.google.inject.Inject;
 import com.google.inject.assistedinject.Assisted;
 
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
-
 /**
  * This is the plugin. Your class should implement one of the existing plugin
  * interfaces. (i.e. AlarmCallback, MessageInput, MessageOutput)
  */
 public class HttpOutput implements MessageOutput {
 
+    public static final int HTTP_BATCH_SIZE = 500;
+    private final OkHttpClient httpClient;
+    private final Gson gson = new Gson();
 	private boolean shutdown;
 	private String url;
 	private static final String CK_OUTPUT_API = "output_api";
+	private static final String CK_GZIP_REQUEST = "gzip_request";
 	private static final Logger LOG = LoggerFactory.getLogger(HttpOutput.class);
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+    /**
+     * This interceptor compresses the HTTP request body. Many webservers can't handle this!
+     *
+     * taken from https://square.github.io/okhttp/interceptors/
+     * */
+    final class GzipRequestInterceptor implements Interceptor {
+        @Override public Response intercept(Interceptor.Chain chain) throws IOException {
+            Request originalRequest = chain.request();
+            if (originalRequest.body() == null || originalRequest.header("Content-Encoding") != null) {
+                return chain.proceed(originalRequest);
+            }
+
+            Request compressedRequest = originalRequest.newBuilder()
+                    .header("Content-Encoding", "gzip")
+                    .method(originalRequest.method(), gzip(originalRequest.body()))
+                    .build();
+            return chain.proceed(compressedRequest);
+        }
+
+        private RequestBody gzip(final RequestBody body) {
+            return new RequestBody() {
+                @Override public MediaType contentType() {
+                    return body.contentType();
+                }
+
+                @Override public long contentLength() {
+                    return -1; // We don't know the compressed length in advance!
+                }
+
+                @Override public void writeTo(BufferedSink sink) throws IOException {
+                    BufferedSink gzipSink = Okio.buffer(new GzipSink(sink));
+                    body.writeTo(gzipSink);
+                    gzipSink.close();
+                }
+            };
+        }
+    }
 
 	@Inject
 	public HttpOutput(@Assisted Stream stream, @Assisted Configuration conf) throws HttpOutputException {
 
-		url = conf.getString(CK_OUTPUT_API);
-		shutdown = false;
+		this.url = conf.getString(CK_OUTPUT_API);
+
+		this.shutdown = false;
 		LOG.info(" Http Output Plugin has been configured with the following parameters:");
 		LOG.info(CK_OUTPUT_API + " : " + url);
```

Comment:
To improve the readability, maybe use `this.url` instead of `url`
```suggestion
		LOG.info(CK_OUTPUT_API + " : " + this.url);
```